xonsh/environ.py

```diff
@@ -61,40 +62,39 @@ def current_branch(cwd=None):
         except subprocess.CalledProcessError:
             pass
 
+    if pad and branch is not None:
+        branch = ' ' + branch
     return branch
 
 
-default_prompt = ('{BOLD_GREEN}{user}@{hostname}{BOLD_BLUE} '
+DEFAULT_PROMPT = ('{BOLD_GREEN}{user}@{hostname}{BOLD_BLUE} '
                   '{cwd}{BOLD_RED}{curr_branch} {BOLD_BLUE}${NO_COLOR} ')
-default_title = '{user}@{hostname}: {cwd} | xonsh'
+DEFAULT_TITLE = '{user}@{hostname}: {cwd} | xonsh'
 
 
-def format_prompt(template=default_prompt):
-    """Formats a xonsh prompt template string.
+def _replace_home(x):
+    return x.replace(builtins.__xonsh_env__['HOME'], '~')
+
+FORMATTER_DICT = dict(user=os.environ.get('USER', '<user>'),
+                      hostname=socket.gethostname().split('.', 1)[0],
+                      cwd=lambda: _replace_home(builtins.__xonsh_env__['PWD']),
+                      curr_branch=lambda: current_branch() or '',
```

Comment:
It seems like this extra level of indirection isn't great.  Instead maybe `current_branch()` should be altered to to return an empty string when not on a branch.  This is non-blocking.